src/backend/distributed/metadata/node_metadata.c

```diff
@@ -724,6 +725,12 @@ master_update_node(PG_FUNCTION_ARGS)
 		LockShardsInPlacementListMetadata(placementList, AccessExclusiveLock);
 	}
 
+	/*
+	 * if we have planned statements such as prepared statements, we should clear the cache so that
+	 * the planned cache doesn't return the old nodename/nodepost.
+	 */
+	ResetPlanCache();
```

Comment:
Is it enough to run this only in this node? Say if worker1 has a plancache and we run master_update_node for worker1, then if we run something directly in worker1 it might actually cause a problem.

I think we should add more tests.

Reply:
In the procedure call delegation / MX case, the sessions on the workers have prepared statements. When a master_update_node happens, we ultimately resync the metadata from the maintendance daemon:
https://github.com/citusdata/citus/blob/master/src/backend/distributed/metadata/metadata_sync.c#L1383

That will effectively drop all the distributed tables and recreate them, which would certainly clear the prepared statement cache.

This area does require more attention, e.g. does this always work correctly in the rebalancer case? (perhaps it does)

Reply:
Should we add rebalancer tests too?